src/full/Agda/TypeChecking/Rewriting.hs

```diff
@@ -292,6 +292,8 @@ checkRewriteRule q = runMaybeT $ setCurrentRange q do
               Function{}     -> usedArgs def
               Axiom{}        -> allVars
               AbstractDefn{} -> allVars
+              Constructor{}  -> allVars
+              Primitive{}    -> allVars
```

Comment:
Would you mind spelling out the catch-all while you are at it?
This would prevent similar regressions in the future.